digiroad2-api-viite/src/main/scala/fi/liikennevirasto/digiroad2/ViiteApi.scala

```diff
@@ -402,10 +405,22 @@ class ViiteApi(val roadLinkService: RoadLinkService, val vVHClient: VVHClient,
         //        roadAddressService.getRoadParts(boundingRectangle, Seq((1, 19999)), municipalities)
         Seq()
       case DrawPublicRoads => roadAddressService.getRoadAddressLinksByLinkId(boundingRectangle, Seq((1, 19999), (40000,49999)), municipalities)
-      case DrawAllRoads => roadAddressService.getRoadAddressLinks(boundingRectangle, Seq(), municipalities, everything = true)
-      case _ => roadAddressService.getRoadAddressLinks(boundingRectangle, Seq((1, 19999)), municipalities)
+      case DrawAllRoads =>
+        val combinedFuture =for{
+          fRoadLink <- Future(roadAddressService.getRoadAddressLinks(boundingRectangle, Seq(), municipalities, everything = true))
```

Comment:
Futures and such should be handled on a service level, API is way too high level (and will mix things)